Segmentation violation of controller and csi-rbdplugin-provisioner with Kubernetes v1.27.1

**Is this a bug report or feature request?**
* Bug Report

**Deviation from expected behavior:**
The rook-ceph-operator and the csi-omap-generator of the csi-rbdplugin-provisioner are crashing after updating my Kubernetes cluster to v1.27.1 because of a "Segmentation violation".

**Expected behavior:**
I expect my Rook-Ceph Setup (v1.11.3) to work with my Kubernetes v1.27.1 cluster, as it should be covered by [Kubernetes version skew policy](https://kubernetes.io/releases/version-skew-policy/). Other containers in the cluster (that also use the Kubernetes API) work just fine after the upgrade.

**How to reproduce it (minimal and precise):**
I just update my Kubernetes v1.26.3 cluster to version 1.27.1 and when I updated the very last node, it stopped working with the errors below. So I'm guessing the reproduction steps would be to just install rook ceph in a v1.27.1 cluster.

I already tried reporting this to [Talos Linux](https://github.com/siderolabs/talos/issues/7110) and [Kubernetes](https://github.com/kubernetes/kubernetes/issues/117466), but they pointed me here. Especially in the [Kubernetes Issue](https://github.com/kubernetes/kubernetes/issues/117466), they referenced https://github.com/kubernetes/kubernetes/pull/116666 and suggested that updating the go dependencies to 1.26.4 would possibly solve this issue.

**Logs to submit**:

rook-ceph-operator (rook-ceph-operator):
```console
2023-04-19 11:08:22.491914 I | rookcmd: starting Rook v1.11.3 with arguments '/usr/local/bin/rook ceph operator'
2023-04-19 11:08:22.491976 I | rookcmd: flag values: --enable-machine-disruption-budget=false, --help=false, --kubeconfig=, --log-level=INFO, --operator-image=, --service-account=
2023-04-19 11:08:22.491979 I | cephcmd: starting Rook-Ceph operator
2023-04-19 11:08:22.706361 I | cephcmd: base ceph version inside the rook operator image is "ceph version 17.2.5 (98318ae89f1a893a6ded3a640405cdbb33e08757) quincy (stable)"
2023-04-19 11:08:22.813675 I | op-k8sutil: ROOK_CURRENT_NAMESPACE_ONLY="false" (env var)
2023-04-19 11:08:22.813690 I | operator: watching all namespaces for Ceph CRs
2023-04-19 11:08:22.813755 I | operator: setting up schemes
2023-04-19 11:08:22.815942 I | operator: setting up the controller-runtime manager
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x12edd10]

goroutine 81 [running]:
k8s.io/client-go/discovery.convertAPIResource(...)
	/home/runner/go/pkg/mod/k8s.io/client-go@v0.26.1/discovery/aggregated_discovery.go:88
k8s.io/client-go/discovery.convertAPIGroup({{{0x0, 0x0}, {0x0, 0x0}}, {{0xc000625068, 0x15}, {0x0, 0x0}, {0x0, 0x0}, ...}, ...})
	/home/runner/go/pkg/mod/k8s.io/client-go@v0.26.1/discovery/aggregated_discovery.go:69 +0x5f0
k8s.io/client-go/discovery.SplitGroupsAndResources({{{0xc000624018, 0x15}, {0xc00015e260, 0x1b}}, {{0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, ...}, ...})
	/home/runner/go/pkg/mod/k8s.io/client-go@v0.26.1/discovery/aggregated_discovery.go:35 +0x2f8
k8s.io/client-go/discovery.(*DiscoveryClient).downloadAPIs(0x25fea6c?)
	/home/runner/go/pkg/mod/k8s.io/client-go@v0.26.1/discovery/discovery_client.go:310 +0x47c
k8s.io/client-go/discovery.(*DiscoveryClient).GroupsAndMaybeResources(0xc0001285a0?)
	/home/runner/go/pkg/mod/k8s.io/client-go@v0.26.1/discovery/discovery_client.go:198 +0x5c
k8s.io/client-go/discovery.ServerGroupsAndResources({0x26299d0, 0xc000904ff0})
	/home/runner/go/pkg/mod/k8s.io/client-go@v0.26.1/discovery/discovery_client.go:392 +0x59
k8s.io/client-go/discovery.(*DiscoveryClient).ServerGroupsAndResources.func1()
	/home/runner/go/pkg/mod/k8s.io/client-go@v0.26.1/discovery/discovery_client.go:356 +0x25
k8s.io/client-go/discovery.withRetries(0x2, 0xc0005a50d0)
	/home/runner/go/pkg/mod/k8s.io/client-go@v0.26.1/discovery/discovery_client.go:621 +0x72
k8s.io/client-go/discovery.(*DiscoveryClient).ServerGroupsAndResources(0x0?)
	/home/runner/go/pkg/mod/k8s.io/client-go@v0.26.1/discovery/discovery_client.go:355 +0x3a
k8s.io/client-go/restmapper.GetAPIGroupResources({0x26299d0?, 0xc000904ff0?})
	/home/runner/go/pkg/mod/k8s.io/client-go@v0.26.1/restmapper/discovery.go:148 +0x42
sigs.k8s.io/controller-runtime/pkg/client/apiutil.NewDynamicRESTMapper.func1()
	/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.14.1/pkg/client/apiutil/dynamicrestmapper.go:86 +0x25
sigs.k8s.io/controller-runtime/pkg/client/apiutil.(*dynamicRESTMapper).setStaticMapper(...)
	/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.14.1/pkg/client/apiutil/dynamicrestmapper.go:119
sigs.k8s.io/controller-runtime/pkg/client/apiutil.NewDynamicRESTMapper(0xc0008eecf0?, {0x0, 0x0, 0x9335f667369e3001?})
	/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.14.1/pkg/client/apiutil/dynamicrestmapper.go:99 +0x176
sigs.k8s.io/controller-runtime/pkg/cluster.setOptionsDefaults.func1(0xc0004f9030?)
	/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.14.1/pkg/cluster/cluster.go:217 +0x25
sigs.k8s.io/controller-runtime/pkg/cluster.New(0xc000888d80, {0xc000999a58, 0x1, 0x0?})
	/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.14.1/pkg/cluster/cluster.go:159 +0x18d
sigs.k8s.io/controller-runtime/pkg/manager.New(_, {0xc0004f9030, 0x0, 0x0, {{0x2623510, 0xc0001234c0}, 0x0}, 0x0, {0x0, 0x0}, ...})
	/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.14.1/pkg/manager/manager.go:351 +0xf9
github.com/rook/rook/pkg/operator/ceph.(*Operator).startCRDManager(0xc000591260, {0x261fbc0?, 0xc000123f00}, 0xc0000c6c60?)
	/home/runner/work/rook/rook/pkg/operator/ceph/cr_manager.go:164 +0x275
created by github.com/rook/rook/pkg/operator/ceph.(*Operator).runCRDManager
	/home/runner/work/rook/rook/pkg/operator/ceph/operator.go:141 +0x185
```

csi-rbdplugin-provisioner (csi-omap-generator):
```console
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x12c9170]

goroutine 1 [running]:
k8s.io/client-go/discovery.convertAPIResource(...)
	/go/src/github.com/ceph/ceph-csi/vendor/k8s.io/client-go/discovery/aggregated_discovery.go:88
k8s.io/client-go/discovery.convertAPIGroup({{{0x0, 0x0}, {0x0, 0x0}}, {{0xc0004a7128, 0x15}, {0x0, 0x0}, {0x0, 0x0}, ...}, ...})
	/go/src/github.com/ceph/ceph-csi/vendor/k8s.io/client-go/discovery/aggregated_discovery.go:69 +0x5f0
k8s.io/client-go/discovery.SplitGroupsAndResources({{{0xc0004a64f8, 0x15}, {0xc000504080, 0x1b}}, {{0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, ...}, ...})
	/go/src/github.com/ceph/ceph-csi/vendor/k8s.io/client-go/discovery/aggregated_discovery.go:35 +0x2f8
k8s.io/client-go/discovery.(*DiscoveryClient).downloadAPIs(0x2036cf1?)
	/go/src/github.com/ceph/ceph-csi/vendor/k8s.io/client-go/discovery/discovery_client.go:310 +0x47c
k8s.io/client-go/discovery.(*DiscoveryClient).GroupsAndMaybeResources(0xc00031e280?)
	/go/src/github.com/ceph/ceph-csi/vendor/k8s.io/client-go/discovery/discovery_client.go:198 +0x5c
k8s.io/client-go/discovery.ServerGroupsAndResources({0x2527b98, 0xc000b83260})
	/go/src/github.com/ceph/ceph-csi/vendor/k8s.io/client-go/discovery/discovery_client.go:392 +0x59
k8s.io/client-go/discovery.(*DiscoveryClient).ServerGroupsAndResources.func1()
	/go/src/github.com/ceph/ceph-csi/vendor/k8s.io/client-go/discovery/discovery_client.go:356 +0x25
k8s.io/client-go/discovery.withRetries(0x2, 0xc0005a5040)
	/go/src/github.com/ceph/ceph-csi/vendor/k8s.io/client-go/discovery/discovery_client.go:621 +0x72
k8s.io/client-go/discovery.(*DiscoveryClient).ServerGroupsAndResources(0x0?)
	/go/src/github.com/ceph/ceph-csi/vendor/k8s.io/client-go/discovery/discovery_client.go:355 +0x3a
k8s.io/client-go/restmapper.GetAPIGroupResources({0x2527b98?, 0xc000b83260?})
	/go/src/github.com/ceph/ceph-csi/vendor/k8s.io/client-go/restmapper/discovery.go:148 +0x42
sigs.k8s.io/controller-runtime/pkg/client/apiutil.NewDynamicRESTMapper.func1()
	/go/src/github.com/ceph/ceph-csi/vendor/sigs.k8s.io/controller-runtime/pkg/client/apiutil/dynamicrestmapper.go:94 +0x25
sigs.k8s.io/controller-runtime/pkg/client/apiutil.(*dynamicRESTMapper).setStaticMapper(...)
	/go/src/github.com/ceph/ceph-csi/vendor/sigs.k8s.io/controller-runtime/pkg/client/apiutil/dynamicrestmapper.go:130
sigs.k8s.io/controller-runtime/pkg/client/apiutil.NewDynamicRESTMapper(0xc000c2e510?, {0x0, 0x0, 0x9335f667369e3001?})
	/go/src/github.com/ceph/ceph-csi/vendor/sigs.k8s.io/controller-runtime/pkg/client/apiutil/dynamicrestmapper.go:110 +0x182
sigs.k8s.io/controller-runtime/pkg/cluster.setOptionsDefaults.func1(0x0?)
	/go/src/github.com/ceph/ceph-csi/vendor/sigs.k8s.io/controller-runtime/pkg/cluster/cluster.go:217 +0x25
sigs.k8s.io/controller-runtime/pkg/cluster.New(0xc0006ae6c0, {0xc000c9f9e8, 0x1, 0x0?})
	/go/src/github.com/ceph/ceph-csi/vendor/sigs.k8s.io/controller-runtime/pkg/cluster/cluster.go:159 +0x18d
sigs.k8s.io/controller-runtime/pkg/manager.New(_, {0x0, 0x0, 0x0, {{0x2522498, 0xc0004b6740}, 0x0}, 0x1, {0x203a454, 0x6}, ...})
	/go/src/github.com/ceph/ceph-csi/vendor/sigs.k8s.io/controller-runtime/pkg/manager/manager.go:359 +0xf9
github.com/ceph/ceph-csi/internal/controller.Start({{0x7ffd3dbaa122, 0x1a}, {0x7ffd3dbaa105, 0x9}, {0x0, 0x0}, 0x0})
	/go/src/github.com/ceph/ceph-csi/internal/controller/controller.go:70 +0x139
main.main()
	/go/src/github.com/ceph/ceph-csi/cmd/cephcsi.go:276 +0x971
```

**Cluster Status to submit**:

* Output of krew commands, if necessary

I can't currently execute the krew commands, because the rook-operator is down, but this is the output from the last time, that I used it before the update:

```
  cluster:
    id:     f781bcf9-2ff7-4775-bd26-e982efedd214
    health: HEALTH_OK
 
  services:
    mon: 3 daemons, quorum a,c,f (age 2m)
    mgr: b(active, since 7w), standbys: a
    osd: 3 osds: 3 up (since 7w), 3 in (since 11w)
 
  data:
    pools:   2 pools, 33 pgs
    objects: 80.75k objects, 310 GiB
    usage:   619 GiB used, 1.3 TiB / 1.9 TiB avail
    pgs:     33 active+clean
 
  io:
    client:   1.2 MiB/s wr, 0 op/s rd, 43 op/s wr
```

Does

**Environment**:
* OS (e.g. from /etc/os-release): Talos Linux v1.4.0
* Kernel (e.g. `uname -a`): 6.1.24-talos
* Cloud provider or hardware configuration: Hetzner Cloud + Dedicated/Bare Metal Servers
* Rook version (use `rook version` inside of a Rook Pod): 1.11.3
* Storage backend version (e.g. for ceph do `ceph -v`): 17.2.5
* Kubernetes version (use `kubectl version`): 1.27.1
* Kubernetes cluster type (e.g. Tectonic, GKE, OpenShift): Vanilla Kubernetes
* Storage backend status (e.g. for Ceph use `ceph health` in the [Rook Ceph toolbox](https://rook.io/docs/rook/latest-release/Troubleshooting/ceph-toolbox/#interactive-toolbox)): I cannot currently check, as the operator is down, but I checked it right before updating and the state was HEALTH_OK with no warnings at all.

--- a/.github/workflows/canary-test-config/action.yaml
+++ b/.github/workflows/canary-test-config/action.yaml
@@ -16,7 +16,7 @@ runs:
     - name: Setup Minikube
       shell: bash --noprofile --norc -eo pipefail -x {0}
       run: |
-        tests/scripts/github-action-helper.sh install_minikube_with_none_driver v1.26.1
+        tests/scripts/github-action-helper.sh install_minikube_with_none_driver v1.27.1
 
     - name: install deps
       shell: bash --noprofile --norc -eo pipefail -x {0}

--- a/.github/workflows/integration-test-helm-suite.yaml
+++ b/.github/workflows/integration-test-helm-suite.yaml
@@ -22,7 +22,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        kubernetes-versions: ["v1.21.14", "v1.26.1"]
+        kubernetes-versions: ["v1.21.14", "v1.27.1"]
     steps:
       - name: checkout
         uses: actions/checkout@v3

--- a/.github/workflows/integration-test-mgr-suite.yaml
+++ b/.github/workflows/integration-test-mgr-suite.yaml
@@ -21,7 +21,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        kubernetes-versions: ["v1.26.1"]
+        kubernetes-versions: ["v1.27.1"]
     steps:
       - name: checkout
         uses: actions/checkout@v3

--- a/.github/workflows/integration-test-multi-cluster-suite.yaml
+++ b/.github/workflows/integration-test-multi-cluster-suite.yaml
@@ -22,7 +22,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        kubernetes-versions: ["v1.26.1"]
+        kubernetes-versions: ["v1.27.1"]
     steps:
       - name: checkout
         uses: actions/checkout@v3

--- a/.github/workflows/integration-test-object-suite.yaml
+++ b/.github/workflows/integration-test-object-suite.yaml
@@ -22,7 +22,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        kubernetes-versions: ["v1.21.14", "v1.26.1"]
+        kubernetes-versions: ["v1.21.14", "v1.27.1"]
     steps:
       - name: checkout
         uses: actions/checkout@v3

--- a/.github/workflows/integration-test-smoke-suite.yaml
+++ b/.github/workflows/integration-test-smoke-suite.yaml
@@ -22,7 +22,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        kubernetes-versions: ["v1.21.14", "v1.26.1"]
+        kubernetes-versions: ["v1.21.14", "v1.27.1"]
     steps:
       - name: checkout
         uses: actions/checkout@v3

--- a/.github/workflows/integration-test-upgrade-suite.yaml
+++ b/.github/workflows/integration-test-upgrade-suite.yaml
@@ -22,7 +22,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        kubernetes-versions: ["v1.21.14", "v1.26.1"]
+        kubernetes-versions: ["v1.21.14", "v1.27.1"]
     steps:
       - name: checkout
         uses: actions/checkout@v3
@@ -67,7 +67,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        kubernetes-versions: ["v1.21.14", "v1.26.1"]
+        kubernetes-versions: ["v1.21.14", "v1.27.1"]
     steps:
       - name: checkout
         uses: actions/checkout@v3

--- a/.github/workflows/integration-tests-on-release.yaml
+++ b/.github/workflows/integration-tests-on-release.yaml
@@ -18,7 +18,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        kubernetes-versions: ["v1.21.14", "v1.23.15", "v1.24.10", "v1.26.1"]
+        kubernetes-versions: ["v1.21.14", "v1.23.15", "v1.24.10", "v1.27.1"]
     steps:
       - name: checkout
         uses: actions/checkout@v3
@@ -59,7 +59,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        kubernetes-versions: ["v1.21.14", "v1.23.15", "v1.24.10", "v1.26.1"]
+        kubernetes-versions: ["v1.21.14", "v1.23.15", "v1.24.10", "v1.27.1"]
     steps:
       - name: checkout
         uses: actions/checkout@v3
@@ -99,7 +99,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        kubernetes-versions: ["v1.21.14", "v1.23.15", "v1.24.10", "v1.26.1"]
+        kubernetes-versions: ["v1.21.14", "v1.23.15", "v1.24.10", "v1.27.1"]
     steps:
       - name: checkout
         uses: actions/checkout@v3
@@ -138,7 +138,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        kubernetes-versions: ["v1.21.14", "v1.23.15", "v1.24.10", "v1.26.1"]
+        kubernetes-versions: ["v1.21.14", "v1.23.15", "v1.24.10", "v1.27.1"]
     steps:
       - name: checkout
         uses: actions/checkout@v3
@@ -177,7 +177,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        kubernetes-versions: ["v1.21.14", "v1.23.10", "v1.26.1"]
+        kubernetes-versions: ["v1.21.14", "v1.23.10", "v1.27.1"]
     steps:
       - name: checkout
         uses: actions/checkout@v3
@@ -220,7 +220,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        kubernetes-versions: ["v1.21.14", "v1.26.1"]
+        kubernetes-versions: ["v1.21.14", "v1.27.1"]
     steps:
       - name: checkout
         uses: actions/checkout@v3

